app/controllers/unattended_controller.rb

```diff
@@ -55,22 +53,26 @@ def hostgroup_template
   # Generate an action for each template kind
   # i.e. /unattended/provision will render the provisioning template for the requesting host
   def host_template
+    return head(:not_found) unless params[:kind].present?
+
+    return render_default_global_template if ipxe_request? && !@host
+    return render_local_boot_template if ipxe_request? && !@host.build?
+
+    return unless verify_found_host
+    return head(:method_not_allowed) unless allowed_to_install?
+    (handle_realm || return) if params[:kind] == 'provision'
+
     render_template params[:kind]
```

Comment:
This is not related but I realized we don't actually enforce `params[:kind]` to be one of known kind strings. I was unable to exploit this tho, to render ERB template from rails I need to pass some slash characters like `views/about/index.html.erb` and even URL encoded ones did not work. We could add it for extra security tho.